src/app/proposals/hooks/useGetProposalsWithGraph.ts

```diff
@@ -1,101 +1,231 @@
+
+  const { data: proposalVotes } = useReadContracts({
+    contracts: votesContracts,
     query: {
-      refetchInterval: AVERAGE_BLOCKTIME,
+      enabled: proposalsFromNode.length > 0,
       staleTime: AVERAGE_BLOCKTIME,
     },
-  })
+  }) as { data?: Array<{ status: string; result: bigint[] }> }
+
+  const { data: quorum } = useReadContracts({
+    contracts: quorumContracts,
+    query: {
+      enabled: proposalsFromNode.length > 0,
+      staleTime: 24 * 60 * 60 * 1000,
+    },
+  }) as { data?: Array<{ status: string; result: bigint }> }
+
+  const { data: state } = useReadContracts({
+    contracts: stateContracts,
+    query: {
+      enabled: proposalsFromNode.length > 0,
+      staleTime: AVERAGE_BLOCKTIME,
+    },
+  }) as { data?: Array<{ status: string; result: bigint }> }
+
+  const blockchainData = useMemo(() => {
+    if (!proposalVotes || !quorum || !state || !proposalsFromNode) return []
+
+    return proposalsFromNode.map((proposal, index) => {
+      const votes = proposalVotes?.[index]?.result?.map(vote => Big(formatEther(vote)).round())
+      const againstVotes = Big(votes?.at(0) ?? 0)
+      const forVotes = Big(votes?.at(1) ?? 0)
+      const abstainVotes = Big(votes?.at(2) ?? 0)
+
+      return {
+        proposalId: proposal.proposalId,
+        votes: {
+          againstVotes,
+          forVotes,
+          abstainVotes,
+        },
+        quorum: Big(formatEther(quorum[index].result).toString()).round(undefined, Big.roundHalfEven),
+        rawState: Big(state?.[index].result?.toString() ?? 0).toNumber() as ProposalState,
+      }
+    })
+  }, [proposalVotes, quorum, state, proposalsFromNode])
 
   const activeProposalCount = useMemo(() => {
     if (!proposalsData || !latestBlockNumber) return '0'
 
-    return proposalsData.proposals
+    return proposalsData
       .filter(
-        (proposal: ProposalGraphQLResponse) =>
+        (proposal: ProposalApiResponse) =>
           handleProposalState(proposal, latestBlockNumber) === ProposalState.Active,
       )
       .length.toString()
   }, [proposalsData, latestBlockNumber])
 
   const totalProposalCount = useMemo(() => {
     if (!proposalsData) return '0'
-    return proposalsData.counters.find(e => e.id === 'proposals')?.count || '0'
+    return proposalsData.length.toString()
   }, [proposalsData])
 
-  const activeProposals: Proposal[] = []
-  const inactiveProposals: Proposal[] = []
-
-  const proposalResponse = proposalsData?.proposals.map(proposal => {
-    const againstVotes = Big(proposal.votesAgainst).div(Big('1e18')).round()
-    const forVotes = Big(proposal.votesFor).div(Big('1e18')).round()
-    const abstainVotes = Big(proposal.votesAbstains).div(Big('1e18')).round()
-    const deadlineBlock = Big(proposal.voteEnd)
-    const creationBlock = Number(proposal.createdAtBlock)
-    const eventArgs = getProposalEventArguments({
-      args: {
+  const { transformedProposals, activeProposals, inactiveProposals } = useMemo(() => {
```

Comment:
This also can be moved to a separate function to keep the code readable.

This hook is already packed with a lot of inline code, that could be converted to functions.